cognitive/src/test/scala/com/microsoft/azure/synapse/ml/services/GlobalParamSuite.scala

```diff
@@ -0,0 +1,50 @@
+// Copyright (C) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in project root for information.
+
+package com.microsoft.azure.synapse.ml.services
+
+import com.microsoft.azure.synapse.ml.Secrets.getAccessToken
+import com.microsoft.azure.synapse.ml.core.test.base.Flaky
+import com.microsoft.azure.synapse.ml.core.test.fuzzing.{TestObject, TransformerFuzzing}
+import com.microsoft.azure.synapse.ml.services.openai.{OpenAIAPIKey, OpenAIPrompt}
+import org.apache.spark.ml.util.MLReadable
+import org.apache.spark.sql.DataFrame
+import org.scalactic.Equality
+
+class GlobalParamSuite extends Flaky with OpenAIAPIKey {
+
+  import spark.implicits._
+
+  override def beforeAll(): Unit = {
+    val aadToken = getAccessToken("https://cognitiveservices.azure.com/")
+    println(s"Triggering token creation early ${aadToken.length}")
+    super.beforeAll()
+  }
+
+  GlobalParams.setDeploymentName(deploymentName)
+
+  lazy val prompt: OpenAIPrompt = new OpenAIPrompt()
+    .setSubscriptionKey(openAIAPIKey)
+    .setCustomServiceName(openAIServiceName)
+    .setOutputCol("outParsed")
+    .setTemperature(0)
+
+  lazy val df: DataFrame = Seq(
+    ("apple", "fruits"),
+    ("mercedes", "cars"),
+    ("cake", "dishes"),
+    (null, "none") //scalastyle:ignore null
+  ).toDF("text", "category")
```

Comment:
We might want to test a few different types of setters in this, 

Consider testing both typed setting and untyped setting of the global param here to ensure all the paths get flexd

Reply:
Added more tests!

Reply:
Added more tests with setting through both the GlobalKey and through a string!